internal/device/energy_zone.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package device
+
+import (
+	"fmt"
+	"math"
+	"sync"
+)
+
+type Zone = string
+
+const (
+	ZonePackage Zone = "package"
+	ZoneCore    Zone = "core"
+	ZoneDRAM    Zone = "dram"
+	ZoneUncore  Zone = "uncore"
+	ZonePSys    Zone = "psys"
+	ZonePP0     Zone = "pp0" // Power Plane 0 - processor cores
+	ZonePP1     Zone = "pp1" // Power Plane 1 - uncore (e.g., integrated GPU)
+)
+
+// zoneKey uniquely identifies a zone by name and index
+type zoneKey struct {
+	name  string
+	index int
+}
+
+// AggregatedZone implements EnergyZone interface by aggregating multiple zones
+// of the same type (e.g., multiple package zones in multi-socket systems).
+// It handles energy counter wrapping for each individual zone and provides
+// a single consolidated energy reading.
+type AggregatedZone struct {
+	name          string
+	index         int
+	zones         []EnergyZone
+	lastReadings  map[zoneKey]Energy
+	currentEnergy Energy // Aggregated energy counter
+	maxEnergy     Energy // Cached sum of all zone MaxEnergy values
+	mu            sync.RWMutex
+}
+
+// NewAggregatedZone creates a new AggregatedZone for zones of the same type
+// The name is taken from the first zone
+// Panics if zones is empty or nil
+func NewAggregatedZone(zones []EnergyZone) *AggregatedZone {
+	// Panic on invalid inputs
+	if len(zones) == 0 {
+		panic("NewAggregatedZone: zones cannot be empty")
+	}
+
+	// Use the first zone's name as the aggregated zone name
+	name := zones[0].Name()
+	// Calculate and cache the combined MaxEnergy during construction
+	// Check for overflow when summing MaxEnergy values
+	var totalMax Energy
+	for _, zone := range zones {
+		zoneMax := zone.MaxEnergy()
+		// Check for overflow before adding
+		if totalMax > 0 && zoneMax > math.MaxUint64-totalMax {
+			// Overflow would occur, use MaxUint64 as safe maximum
+			totalMax = Energy(math.MaxUint64)
+			break
+		}
+		totalMax += zoneMax
+	}
+
+	return &AggregatedZone{
+		name:          name,
+		index:         -1, // Indicates this is an aggregated zone
+		zones:         zones,
+		lastReadings:  make(map[zoneKey]Energy),
+		currentEnergy: 0,
+		maxEnergy:     totalMax, // Cache the combined MaxEnergy
+	}
+}
+
+// Name returns the zone name
+func (az *AggregatedZone) Name() string {
+	return az.name
+}
+
+// Index returns the zone index (-1 for aggregated zones)
+func (az *AggregatedZone) Index() int {
+	return az.index
+}
+
+// Path returns path for the aggregated zone
+func (az *AggregatedZone) Path() string {
+	// TODO: decide if all the paths should be returned
+	return fmt.Sprintf("aggregated-%s", az.name)
+}
```

Comment:
`Path()` returns name?

Reply:
Yes, since we can't point to single path, I thought an easy way to identify is to set Path as aggregated. We definitely can improve this by finding the common part of the parts and replacing the difference with a regex. I will add an issue for us to fix this later.

Reply:
Or do you think appending the parts into a single string is sufficient?

Reply:
Added #2195